### PR TITLE
`Development`: Increase client test coverage for submission export

### PR DESCRIPTION
**Pull request by ole-ve**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation
Increase the code coverage for service and components in `app/exercises/shared/submission-export`.

### Description
Added test cases for both submission export files (name: Branch-Cov, Line-Cov):
- `submission-export-button.component.ts`: 75%, 100%
- `submission-export-dialog.component.ts`: 75%, 88.57%
- `submission-export.service.ts`: 87.5%, 100%

Overall (`app/exercises/shared/submission-export`): 78.57%, 93.54%

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by cremertim — APPROVED** [review]
code changes lgtm. Great work 👍

**Review by willergeorg — APPROVED** [review]
Code changes are fine for me

File: src/test/javascript/spec/component/exercises/shared/submission-export-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        let handleExportResponseMock: jest.Mock;
+
+        beforeEach(() => {
+            handleExportResponseMock = jest.fn().mockReturnValue(of());
+            component.handleExportResponse = handleExportResponseMock;
+        });
+
+        afterEach(() => {
+            jest.restoreAllMocks();
+        });
```
Comment (coaxial1) [inline src/test/javascript/spec/component/exercises/shared/submission-export-dialog.component.spec.ts:111]:
```suggestion
```
This isn't necessary, as you're already doing this in the 'beforeEach' block of the outer 'describe' block

**Review by coaxial1 — CHANGES_REQUESTED** [review]
Looking good! Added two minor suggestions below 👍

File: src/test/javascript/spec/component/exercises/shared/submission-export-button.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { SubmissionExportButtonComponent } from 'app/exercises/shared/submission-export/submission-export-button.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { MockNgbModalService } from '../../../helpers/mocks/service/mock-ngb-modal.service';
+
+export class MockNgbModalRef {
+    componentInstance = {
+        exerciseId: undefined,
+        exerciseType: undefined,
+    };
+    result: Promise<any> = new Promise((resolve, reject) => resolve(true));
```
[the remark this answers is not in the document]
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/submission-export-button.component.spec.ts]: Implemented.

File: src/test/javascript/spec/component/exercises/shared/submission-export-dialog.component.spec.ts (hunk 1 above)
Comment (ole-ve) [reply src/test/javascript/spec/component/exercises/shared/submission-export-dialog.component.spec.ts]: Implemented.

**Review by cremertim — APPROVED** [review]
re-approve after integrating Patricks Feedback 👍

**Review by coaxial1 — APPROVED** [review]
LGTM! 👍

**Review by akesfeden — APPROVED** [review]
Added tests look good, GitHub client tests are passing so I approve.

**Review by willergeorg — APPROVED** [review]

**Pull request merged**